src/app/products/manage-tax-components/create-tax-component/create-tax-component.component.ts

```diff
@@ -0,0 +1,129 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, FormControl } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { ProductsService } from '../../products.service';
+
+/**
+ * Create Tax Component component.
+ */
+@Component({
+  selector: 'mifosx-create-tax-component',
+  templateUrl: './create-tax-component.component.html',
+  styleUrls: ['./create-tax-component.component.scss']
+})
+export class CreateTaxComponentComponent implements OnInit {
+
+  /** Minimum start date allowed. */
+  minDate = new Date();
+  /** Maximum start date allowed. */
+  maxDate = new Date(2025, 6, 22);
+  /** Tax Component form. */
+  taxComponentForm: FormGroup;
+  /** Tax Component template data. */
+  taxComponentTemplateData: any;
+  /** Credit Account Type data. */
+  creditAccountTypeData: any;
+  /** Credit Account data. */
+  creditAccountData: any;
+  /** Debit Account Type data. */
+  debitAccountTypeData: any;
+  /** Debit Account data. */
+  debitAccountData: any;
+
+  /**
+   * Retrieves the tax Component template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ProductsService} productsService Products Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {DatePipe} datePipe Date Pipe to format date.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private productsService: ProductsService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe(( data: { taxComponentTemplate: any }) => {
+      this.taxComponentTemplateData = data.taxComponentTemplate;
+    });
+  }
+
+  /**
+   * Creates the tax Component form
+   */
+  ngOnInit() {
+    this.createTaxComponentForm();
+    this.setConditionalControls();
+  }
+
+  /**
+   * Creates the tax Component form
+   */
+  createTaxComponentForm() {
+    this.creditAccountTypeData = this.debitAccountTypeData = this.taxComponentTemplateData.glAccountTypeOptions;
+    this.taxComponentForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'percentage': ['', [Validators.required, Validators.min(1), Validators.max(100)]],
+      'creditAccountType': [''],
+      'debitAccountType': [''],
+      'startDate': ['', Validators.required],
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the tax Component form
+   */
+  setConditionalControls() {
+    this.taxComponentForm.get('debitAccountType').valueChanges.subscribe(debitAccountTypeId => {
+      this.debitAccountData = this.getAccountsData(debitAccountTypeId);
+      this.taxComponentForm.addControl('debitAcountId', new FormControl('', Validators.required));
```

Comment:
```suggestion
      this.taxComponentForm.addControl('debitAccountId', new FormControl('', Validators.required));
```

Reply:
@punwai That's not a typo that's the current fineract API implementation

Reply:
@OussEmaDevCode My mistakes

Reply:
This is not an issue, it is the API format. The typo needs to be corrected at the server end first.